### PR TITLE
improve performance of parseRange

### PR DESCRIPTION
**Pull request by Uzlopak**
before:

```sh
node benchmarks/parseRange.js
size: 150, bytes=0-100 x 1,905,512 ops/sec ±0.40% (191 runs sampled)
size: 150, bytes=0-4,90-99,5-75,100-199,101-102 x 490,499 ops/sec ±0.32% (191 runs sampled)
```

after:

```sh
node benchmarks/parseRange.js
size: 150, bytes=0-100 x 12,073,083 ops/sec ±0.48% (190 runs sampled)
size: 150, bytes=0-4,90-99,5-75,100-199,101-102 x 1,380,042 ops/sec ±0.48% (193 runs sampled)
```

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Can you keep the range unit?
Although there are only `bytes` is usable in https://www.iana.org/assignments/http-parameters/http-parameters.xhtml#range-units. It also own `none` for non supported range.

[RFC 9110](https://httpwg.org/specs/rfc9110.html#considerations.for.new.range.units) also not block for custom specify unit.

**Comment by Uzlopak** [comment]
@climba03003 There is a regex check so that only ranges with "bytes=" were processed. So if "none" is set, it wont be processed by parseRange.

So we can savely drop the type as we never process the "none"-case?!

**Comment by climba03003** [comment]
> There is a regex check so that only ranges with "bytes=" were processed. So if "none" is set, it wont be processed by parseRange.

Yes, the `.type` is not used in anywhere. 
But it is more complete for the function itself to provide the whole view of `Range` header.

File: lib/SendStream.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ function SendStream (req, path, options) {
   this._maxage = typeof this._maxage === 'string'
     ? ms.parse(this._maxage)
     : Number(this._maxage)
-  this._maxage = !isNaN(this._maxage)
+  this._maxage = this._maxage === this._maxage // eslint-disable-line no-self-compare
```
Comment (climba03003) [inline lib/SendStream.js:135]: Can you add a explanation for each changes.
It is not obvious to know what it happen in first view.
Something like `fast path of isNaN(number)`

**Comment by Uzlopak** [comment]
Can I just simply rename it to parseBytesRange?

**Comment by climba03003** [comment]
> Can I just simply rename it to parseBytesRange?

Yes

**Comment by Uzlopak** [comment]
@climba03003 
PTAL

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — APPROVED** [review]

**Pull request merged**